apps/src/gamelab/animationListModule.js

```diff
@@ -300,17 +353,28 @@ export function addAnimation(key, props) {
   // TODO: Validate that key is not already in use?
   // TODO: Validate props format?
   return (dispatch, getState) => {
-    dispatch({
-      type: ADD_ANIMATION,
-      key,
-      props
-    });
+    props.looping = true;
+    dispatch(addAnimationAction(key, props));
     dispatch(loadAnimationFromSource(key, () => {
       dispatch(selectAnimation(key));
     }));
     let name = generateAnimationName(props.name, getState().animationList.propsByKey);
     dispatch(setAnimationName(key, name));
-    dashboard.project.projectChanged();
+    projectChanged();
+  };
+}
+
+/**
+ * Append an animation to the project (at the end of the list of frames).
+ * @param {!SerializedAnimation} props
+ * @returns {function}
+ */
+export function appendCustomFrames(props) {
+  return (dispatch, getState) => {
```

Comment:
nit: in cases where you don't actually use getState at all, i'd just make this `return dispatch => {`